onnxruntime/core/optimizer/unsqueeze_elimination.cc

```diff
@@ -40,13 +40,17 @@ Status UnsqueezeElimination::Apply(Graph& graph, Node& node, RewriteRuleEffect&
   // Generate new dims.
   InlinedVector<int64_t> new_dims(output_rank, 0);
   for (int64_t axis : axes) {
-    new_dims[static_cast<size_t>(axis)] = 1;
+    if (static_cast<size_t>(axis) < new_dims.size()) {
```

Comment:
Could the axis be negative or not?
In which case it could be bigger than new_dims.size()? Maybe it's better to throw an exception instead of ignoring the problem.
Who was the original author of this code? We should ask him/her.

Reply:
Right above this, it checks for negative axis. 
@hariharans29 Can you please confirm whether it's better to throw an exception here?

Reply:
In this case the bug, tensor_proto.dims().size() = 0 and axes = [1, 2] and caused memory access error